core/src/main/scala/org/apache/spark/ui/WebUI.scala

```diff
@@ -95,6 +101,15 @@ private[spark] abstract class WebUI(
     serverInfo.foreach(_.addHandler(handler, securityManager))
   }
 
+  /** Attaches a handler to this UI. */
+  def attachHandler(contextPath: String, httpServlet: HttpServlet, pathSpec: String): Unit =
+    synchronized {
```

Comment:
Does this need to be synchronized? the `attachHandler` it calls already is and the rest doesn't look like it needs it.

Reply:
You are right. updated